packages/blob-storage/internal/indexer/indexer.go

```diff
@@ -0,0 +1,94 @@
+package indexer
+
+import (
+	"context"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// Indexer struct holds the configuration and state for the Ethereum chain listener.
+type Indexer struct {
+	rpcURL            string
+	beaconURL         string
+	networkName       string
+	contractAddress   common.Address
+	eventHash         common.Hash
+	startHeight       *big.Int
+	eventCallbackFunc func(string, string, string, types.Log)
```

Comment:
nondescriptive as to what the three things are - you could provide named parameters here